### DISCUSSION TITLE
Multiple Simultaneous Dialogs Mode

### DISCUSSION THREAD

**Discussion by mjacoup** [serenity-is/Serenity#5509] [Help]
Thanks Volkanceylan for this great platform 
A client asked us to enable open more than windows (dialogs) in the same time and can minimize , activate and activate any window ( dialog) because they are currently make this by open tabs in browsers . I see it is logic requirement and wonder is serenity handle this mode in basic or advanced samples ? if no how can i do this ?  

Thanks in advance

**Comment by VictorTomaili** [comment]
Multiple dialog is possible but we don't have samples for that. And it's brings some of problems together. For example, what happens on browser refresh, also what happens when clicked on different menu item or some browser navigation event. When user miss click to somewhere, all dialogs will be lost. This feature may require single page application and url handling together. Or you need handle state of changes per dialog and bring it back to dialogs on refresh or navigate. Serenity has all of features for save states, or sample for prevent lost changes. But also you need handle windows for minimize, or bring to front when clicked background dialog. There is no sample about multiple windows can be used same time.

**Reply by mjacoup** [reply]
Thanks for reply 
1- what happens when clicked on different menu item :--> in our case open the new item in new dialog and so on .
2- On browser refresh --> give warning if unsaved changes 
3- This feature may require single page application : yes it will be more effective with SPA but is it not exist with reactJs version in serenity ?

**Reply by VictorTomaili** [reply]
serenity uses react for components mostly. But there is no restriction to use it as SPA. You have to do a lot of task if you want to do this. Have a good luck
